src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/pyrometer/Api.java

```diff
@@ -0,0 +1,152 @@
+ *  A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.pyrometer;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.temperature.CompactNodeSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.temperature.ShardProfileSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.temperature.HeatZoneAssigner;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.temperature.TemperatureVector;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+
+/**
+ * Although Pyrometer exists on all the nodes - data and master but the APIs are only accessible
+ * from the master nodes. Pyrometer is able answer queries such as:
+ * - All nodes by in Zone <i>Z</i> by dimension <i>D</i> order by <i>Temperature</i> in
+ * descending order.
+ * - The top <i>K</i> warm shards on the hottest nodes by dimension D over a period of h hours
+ * -
+ */
+public class Api {
+    enum Count {
+        /**
+         * This is to ask the API to return all the contenders.
+         */
+        ALL,
+
+        /**
+         * This is to return the top K contenders
+         */
+        TOP_K
+    }
+
+    enum SortOrder {
+        ASCENDING,
+        DESCENDING;
+    }
+
+    /**
+     * Api class should be called only from elected master node.
+     *
+     * @return true if elected master or false otherwise.
+     */
+    private static boolean checkIfElectedMaster() {
+        return true;
+    }
+
+    private static void failIfNotElectedMaster(String apiName) {
+        if (!checkIfElectedMaster()) {
+            StringBuilder builder = new StringBuilder();
+            builder.append("Api (").append(apiName).append(") can only be called from the elected"
+                    + " master node.");
+
+            throw new IllegalStateException(builder.toString());
+        }
+    }
+
+    /**
+     * A cluster is considered imbalanced along a dimension if the temperature of the hottest
+     * node and the coldest node are too far apart. How far is too far, is something to be
+     * evaluated.
+     *
+     * @param dimension The dimension to consider.
+     * @return The difference in temperature of the
+     */
+    public static boolean isClusterImbalanceAlongDimension(TemperatureVector.Dimension dimension) {
+        failIfNotElectedMaster("getClusterImbalanceAlongDimension");
+        // TODO: Calculate this
```

Comment:
A metric here would also be helpful.

Reply:
These are just placeholder methods that can be called by the component dependent on them. We can talk about metrics when we implement them.